aws/table_aws_elasticache_cluster.go

```diff
@@ -0,0 +1,281 @@
+				Transform:   transform.FromField("TagList").Transform(clusterTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ARN").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listElasticCacheClusters(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	// Create Session
+	svc, err := ElasticacheService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	err = svc.DescribeCacheClustersPages(
+		&elasticache.DescribeCacheClustersInput{},
+		func(page *elasticache.DescribeCacheClustersOutput, isLast bool) bool {
+			for _, cacheCluster := range page.CacheClusters {
+				d.StreamListItem(ctx, cacheCluster)
+			}
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getElasticCacheCluster(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	// create service
+	svc, err := ElasticacheService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	quals := d.KeyColumnQuals
+	cacheClusterID := quals["cache_cluster_id"].GetStringValue()
+
+	params := &elasticache.DescribeCacheClustersInput{
+		CacheClusterId: aws.String(cacheClusterID),
+	}
+
+	op, err := svc.DescribeCacheClusters(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if op.CacheClusters != nil && len(op.CacheClusters) > 0 {
+		return op.CacheClusters[0], nil
+	}
+	return nil, nil
+}
+
+func listTagsForElasticacheCluster(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("ListTagsForElasticacheCluster")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	cluster := h.Item.(*elasticache.CacheCluster)
+
+	// Create session
+	svc, err := ElasticacheService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build param
+	param := &elasticache.ListTagsForResourceInput{
+		ResourceName: cluster.ARN,
+	}
+
+	clusterTags, err := svc.ListTagsForResource(param)
+
+	if err != nil {
+		return nil, err
+	}
+	return clusterTags, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func clusterTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("clusterTagListToTurbotTags")
+	clusterTag := d.HydrateItem.(*elasticache.TagListMessage)
+
```

Comment:
Return `nil`, if `clusterTag.TagList` value is nil.
```suggestion

if clusterTag.TagList == nil {
  return nil, nil
}

```